### PR TITLE
Implement talk page for all articles

### PR DESCRIPTION
**Pull request by cooltey**
https://phabricator.wikimedia.org/T266169

### CODE REVIEW

File: app/src/main/java/org/wikipedia/page/PageFragment.java [hunk 1]
```diff
@@ -1195,6 +1189,11 @@ public void startDescriptionEditActivity(@Nullable String text) {
         }
     }
 
+    private void startTalkTopicActivity(@NonNull PageTitle pageTitle) {
+        // If we're currently looking at a User page, then go directly to the corresponding User Talk page.
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/page/PageFragment.java:1193]: This comment is no longer relevant here.

**Review by dbrant — CHANGES_REQUESTED** [review]
This is pretty much ready to go, just some minor nits.

File: app/src/main/java/org/wikipedia/talk/TalkTopicActivity.kt [hunk 2]
```diff
@@ -60,19 +61,16 @@ class TalkTopicActivity : BaseActivity(), LinkPreviewDialog.Callback {
         title = ""
         linkHandler = TalkLinkHandler(this)
 
-        if (intent.hasExtra(EXTRA_LANGUAGE)) {
-            wikiSite = WikiSite.forLanguageCode(intent.getStringExtra(EXTRA_LANGUAGE).orEmpty())
-        }
-        userName = intent.getStringExtra(EXTRA_USER_NAME).orEmpty()
+        pageTitle = GsonUnmarshaller.unmarshal(PageTitle::class.java, intent.getStringExtra(EXTRA_PAGE_TITLE))
```
[the remark this answers is not in the document]
Comment (cooltey) [reply app/src/main/java/org/wikipedia/talk/TalkTopicActivity.kt]: Done.

File: app/src/main/java/org/wikipedia/page/PageFragment.java (hunk 1 above)
Comment (cooltey) [reply app/src/main/java/org/wikipedia/page/PageFragment.java]: Done.

**Pull request merged**